module/support/optimisation/NSGA2Optimiser/src/tasks/MultiPathOptimiser.cpp

```diff
@@ -0,0 +1,165 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022 NUbots
+ *
+ * This file is part of the NUbots codebase.
+ * See https://github.com/NUbots/NUbots for further info.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#include "MultiPathOptimiser.hpp"
+
+#include <fstream>  // ifstream
+#include <nuclear>
+#include <sstream>  //istringstream
+
+#include "nsga2/NSGA2.hpp"
+
+#include "extension/Configuration.hpp"
+
+#include "message/support/optimisation/NSGA2Evaluator.hpp"
+#include "message/support/optimisation/NSGA2Optimiser.hpp"
+
+#include "utility/support/yaml_expression.hpp"
+
+namespace module::support::optimisation {
+    using message::support::optimisation::NSGA2EvaluationRequest;
+    using utility::support::Expression;
+
+    void MultiPathOptimiser::setup_nsga2(const ::extension::Configuration& config, nsga2::NSGA2& nsga2_algorithm) {
+        NUClear::log<NUClear::INFO>("Multipath Optimiser Setting up NSGA2");
+        std::vector<std::vector<double>> population_data;
+
+        std::string input_file_name =
+            "../NUbots/module/support/optimisation/NSGA2Optimiser/data/config/Supplied_Pop_Reals.csv";
```

Comment:
yikes

Reply:
An on<Trigger<CommandLineArguments>> would probably be ideal when used in the relevant reactor? Or just a config coming from some yaml file during on<Configuration>